kngsild/src/main/kotlin/io/egm/kngsild/utils/AuthUtils.kt

```diff
@@ -15,51 +13,65 @@ import java.net.http.HttpRequest
 import java.net.http.HttpResponse
 
 class AuthUtils(
-    private val serverUrl: String,
-    private val clientId: String,
-    private val clientSecret: String,
-    private val grantType: String,
-    private val enabled: Boolean = true
+    private val clientCredentials: ClientCredentials? = null,
+    private val providedToken: ProvidedToken? = null,
+    private val authType: AuthType = AuthType.NONE
 ) {
 
+    init {
+        when (authType) {
+            AuthType.PROVIDED_TOKEN ->
+                if (providedToken == null) throw ConfigurationError("ProvidedToken are not set")
```

Comment:
you could set more human friendly error messages (here and 2 lines below). Someone who has not developed the lib will never understand what the error is about.